packages/rulesets/src/spectral/test/put-request-response-scheme-arm.test.ts

```diff
@@ -0,0 +1,641 @@
+  })
+})
+
+test("PutRequestResponseScheme should find no errors", () => {
+  const myOpenApiDocument = {
+    swagger: "2.0",
+    paths: {
+      "/api/configServers": {
+        put: {
+          operationId: "ConfigServers_Update",
+          parameters: [
+            {
+              $ref: "#/parameters/ApiVersionParameter",
+            },
+            {
+              $ref: "#/parameters/SubscriptionIdParameter",
+            },
+            {
+              $ref: "#/parameters/PathResourceParameter",
+            },
+          ],
+          responses: {
+            "200": {
+              description: "Success",
+              schema: {
+                $ref: "#/definitions/ConfigServerResource",
+              },
+            },
+          },
+        },
+      },
+    },
+    definitions: {
+      ConfigServerResource: {
+        description: "Config Server resource",
+        type: "object",
+        properties: {
+          provisioningState: {
+            description: "State of the config server.",
+            enum: ["NotAvailable", "Deleted", "Failed", "Succeeded", "Updating"],
+            type: "string",
+            readOnly: true,
+            "x-ms-enum": {
+              name: "ConfigServerState",
+              modelAsString: true,
+            },
+          },
+        },
+      },
+      ConfigServerResources: {
+        description: "Config Server resources",
+        type: "object",
+        properties: {
+          provisioningStates: {
+            description: "State of the config server.",
+            enum: ["NotAvailable", "Deleted", "Failed", "Succeeded", "Updating"],
+            type: "string",
+            readOnly: true,
+            "x-ms-enum": {
+              name: "ConfigServerState",
+              modelAsString: true,
+            },
+          },
+        },
+      },
+    },
+    parameters: {
+      ApiVersionParameter: {
+        name: "api-version",
+        in: "query",
+        description: "The API version to use for this operation.",
+        required: true,
+        type: "string",
+        minLength: 1,
+      },
+      SubscriptionIdParameter: {
+        name: "subscriptionId",
+        in: "path",
+        description:
+          "Gets subscription ID which uniquely identify the Microsoft Azure subscription. The subscription ID forms part of the URI for every service call.",
+        required: true,
+        type: "string",
+      },
+      PathResourceParameter: {
+        name: "pathResource",
+        in: "body",
+        description: "Parameters for the update operation",
+        required: true,
+        schema: {
+          $ref: "#/definitions/ConfigServerResource",
+        },
+      },
+    },
+  }
+  return linter.run(myOpenApiDocument).then((results) => {
+    expect(results.length).toBe(0)
+  })
+})
+
+test("PutRequestResponseScheme for json property with empty value in swagger should find no errors. JobAgent.x-ms-arm-id-details is empty below", () => {
```

Comment:
Can we remove extra fields which are not required for this test, it's hard to read.